### PR TITLE
Add region to pipeline bootstrap interactive flow

### PR DESCRIPTION
**Pull request by aahung**
#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/pipeline/bootstrap/guided_context.py [hunk 1]
```diff
@@ -40,6 +42,13 @@ def run(self) -> None:
                 type=click.STRING,
             )
 
+        if not self.region:
+            self.region = click.prompt(
+                "\nAWS region (the AWS region where the environment infrastructure resources will be deployed to)",
+                type=click.STRING,
+                default="us-east-1",
```
Comment (c2tarun) [inline samcli/commands/pipeline/bootstrap/guided_context.py:49]: should we default to `us-east-1` or boto.session.region?

**Review by c2tarun — CHANGES_REQUESTED** [review]

**Review by hoffa — APPROVED** [review]
LGTM. Not sure about the newline but we can revisit.

File: samcli/commands/pipeline/bootstrap/guided_context.py (hunk 1 above)
Comment (aahung) [reply samcli/commands/pipeline/bootstrap/guided_context.py]: updated, good catch

**Comment by aahung** [comment]
> LGTM. Not sure about the newline but we can revisit.

It was to be consistent with others, we should revisit

We don't need to adjust our templates because `region` is already configured to read from the toml file

image

**Review by c2tarun — APPROVED** [review]

**Pull request merged**